implementations/rust/ockam/ockam_command/src/zone/common_args.rs

```diff
@@ -143,13 +144,57 @@ impl ZoneNameLongOrConfigArg {
 
 #[derive(Clone, Debug, Args, Default)]
 pub struct SecretsConfigArg {
-    /// The path to the secrets file, in yaml or json format.
+    /// The path to the secrets file, in yaml or json format, or the inline content of the secrets.
     /// If not set, the `./secrets.yaml` file from the current directory will be used.
-    /// If no file is found, the command will just list the existing secrets.
-    #[arg(long, visible_alias = "secrets")]
+    #[arg(long, visible_alias = "secrets", env = "OCKAM_ZONE_SECRETS")]
     pub secrets_config: Option<String>,
 }
 
+impl SecretsConfigArg {
+    /// Get the contents of the secrets config as a string.
+    /// If the input is a file path that exists, read from the file.
+    /// Otherwise, treat the input as inline content.
+    pub fn contents(&self) -> crate::Result<Option<String>> {
+        let config_input = match &self.secrets_config {
+            Some(input) => input,
+            None => {
+                return if std::path::Path::new("./secrets.yaml")
+                    .try_exists()
+                    .into_diagnostic()?
+                {
+                    Ok(Some(
+                        std::fs::read_to_string("./secrets.yaml")
+                            .into_diagnostic()
+                            .wrap_err("Failed to read ./secrets.yaml")?,
+                    ))
+                } else if std::path::Path::new("./secrets.yml")
+                    .try_exists()
+                    .into_diagnostic()?
+                {
+                    Ok(Some(
+                        std::fs::read_to_string("./secrets.yml")
+                            .into_diagnostic()
+                            .wrap_err("Failed to read ./secrets.yml")?,
+                    ))
+                } else {
+                    Ok(None)
+                }
+            }
+        };
+
+        // Try to read as file first, if it fails treat as inline content
+        let contents = if std::path::Path::new(config_input).exists() {
+            std::fs::read_to_string(config_input)
+                .into_diagnostic()
+                .wrap_err(format!("Failed to read secrets file at {}", config_input))?
```

Comment:
"Failed to read the secrets definition" maybe since this is not a file in that case?

Reply:
Correct, I'll fix it.